src/geometry/LinearPropagator.hh

```diff
@@ -23,24 +24,66 @@ class LinearPropagator
     struct result_type
     {
         real_type distance; //!< Distance traveled
-        VolumeId  volume;   //!< Post-propagation volume
+        bool      boundary; //!< True if hit a boundary before given distance
     };
 
   public:
-    //! Construct from persistent and state data
+    // Construct from persistent and state data
     inline CELER_FUNCTION LinearPropagator(GeoTrackView* track);
 
-    //! Move track to next volume boundary.
+    // Move track to next volume boundary.
     inline CELER_FUNCTION result_type operator()();
 
-    //! Move track by a user-provided distance, or to the next boundary
+    // Move track by a user-provided distance, or to the next boundary
     inline CELER_FUNCTION result_type operator()(real_type dist);
 
   private:
     GeoTrackView& track_;
 };
 
 //---------------------------------------------------------------------------//
-} // namespace celeritas
+/*!
+ * Construct from persistent and state data.
+ */
+CELER_FUNCTION LinearPropagator::LinearPropagator(GeoTrackView* track)
+    : track_(*track)
+{
+    CELER_EXPECT(track);
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Move track to next volume boundary.
+ */
+CELER_FUNCTION
+LinearPropagator::result_type LinearPropagator::operator()()
+{
+    result_type result;
+    result.distance = track_.move_to_boundary();
+    result.boundary = true;
+    return result;
+}
 
-#include "LinearPropagator.i.hh"
+//---------------------------------------------------------------------------//
+/*!
+ * Move track by a user-provided distance up to the next boundary.
+ */
+CELER_FUNCTION
+LinearPropagator::result_type LinearPropagator::operator()(real_type dist)
+{
+    CELER_EXPECT(dist > 0);
+
+    if (dist >= track_.next_step())
+    {
+        // Move to boundary
+        return (*this)();
+    }
+
+    result_type result;
+    result.distance = track_.move_internal(dist);
+    result.boundary = false;
+    return resuult;
```

Comment:
Typo here.

Reply:
Wow, I'm surprised this compiled. Good catch!

Reply:
Actually it looks like you might've mistakenly reviewed a single commit (which contained the typo) rather than the final changeset. Can you look at https://github.com/celeritas-project/celeritas/pull/333/files ?